docs/source/using/host-folders.adoc

```diff
@@ -0,0 +1,225 @@
+
+The following sections describe the prerequisites for using the `hostfolder` command, as well as the various sub-commands it provides.
+
+[[host-folder-prerequisites]]
+=== Prerequisites
+
+The prerequisite for using the `hostfolder` command is the ability to share directories via CIFS.
+On Windows CIFS is the default technology for sharing directories.
+For example, on Windows 10 the _C:\Users_ folder is shared by default and can be accessed by locally-authenticated users.
+
+It is also possible to use CIFS on OS X and Linux.
+On the former you can enable CIFS based shares via the  _System Preferences > Sharing_.
+Refer to link:https://support.apple.com/en-us/HT204445[How to connect with File Sharing on your Mac] for detailed setup instructions.
+
+On Linux follow your distribution specific instructions to install link:https://www.samba.org[Samba].
+
+[[displaying-host-folders]]
+=== Displaying host folders
+
+The first `hostfolder` command is the `list` command.
+It gives you an overview of the defined host folders, their name, mount point, remote path and whether they are currently mounted.
+An example output could look like:
+
+----
+$ minishift hostfolder list
+Name    Mountpoint        Remote path          Mounted
+mac     /mnt/sda1/mac     //192.168.1.82/MAC   N
+----
+
+In this example, there is a host folder with the name _mac_ which mounts _//192.168.1.82/MAC_ onto _/mnt/sda1/mac_ in the Minishift VM.
+The share is currently not mounted.
+
+NOTE: The remote path must be reachable from within the VM.
+In the example above _192.168.1.82_ is the IP of host within the LAN which is one options one can use.
+You can use `ifconfig` (or `Get-NetIPAddress | Format-Table` on Windows) to determine a routable IP address.
+
+[[adding-host-folders]]
+=== Adding host folders
+
+The link:../command-ref/minishift_hostfolder_add{outfilesuffix}[`hostfolder add`] command allows you define a new host folder.
+This in an interactive process which queries the relevant details for a host folder based on CIFS.
+<<adding-cifs-hostfolder>> shows an example.
+
+[[adding-cifs-hostfolder]]
+.Adding a CIFS based hostfolder
+----
+$ minishift hostfolder add mac // <1>
+UNC path: //192.168.99.1/MAC   // <2>
+Mountpoint [/mnt/sda1/mac]:    // <3>
+Username: john                 // <4>
+Password: [HIDDEN]             // <5>
+Domain:                        // <6>
+Added: mac
+----
+<1> (Required) Actual `hostfolder add` command, specifying a hostfolder with a name of _mac_.
+<2> (Required) The UNC path for the share.
+<3> The mount point within the VM. The default is _/mnt/sda1/<host folder name>_.
+<4> (Required) The username for the CIFS share.
+<5> (Required) The password for the CIFS share.
+<6> The domain of the share. Often this can be left blank, but for example on Windows, when your account is linked to a Microsoft account, you must use the Microsoft account email address as username as well as your machine name as displayed by _$env:COMPUTERNAME_ as domain.
```

Comment:
It is mainly about the domain. It seems tricky to get it into the right scope using a NOTE at this point. It does not nest the way I'd like.